fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/metadata/expressions/KeyExpression.java

```diff
@@ -207,6 +207,18 @@ default boolean hasLosslessNormalization() {
         return !createsDuplicates();
     }
 
+    /**
+     * Method that indicates whether the keys can be stored in a fixed width.  For example, an integer or long can be
+     * stored as a fixed width byte[].
+     *
+     * This is utilized by Lucene to see if we can optimize the key lookup in the case of deletes.
+     *
+     */
+    @API(API.Status.INTERNAL)
+    default boolean isFixedWidth(Descriptors.Descriptor descriptor) {
+        return false;
```

Comment:
You will also encounter things like a primary key [component] in a nested "header-like" field.

Reply:
Concatenated keys are the other big miss here

Reply:
I guess grouping keys also need override for this probably?